File: java/sodium/src/sodium/Listener.java
package sodium;

import java.util.List;
import java.util.ArrayList;

public class Listener {
    public Listener() {}

    public void unlisten() {}

    /**
     * Combine listeners into one where a single unlisten() invocation will unlisten
     * both the inputs.
     */
    public Listener append(final Listener l) {
        return new Listener() {
            public void unlisten() {
                this.unlisten();
                l.unlisten();
            }
        };
    }
}

File: c#/Sodium/src/Sodium/Listener.cs
using System;

namespace Sodium
{
  public class Listener
  {
    public Listener()
    {
      Unlisten = () => { };
    }
    public Action Unlisten { get; protected set; }

    public Listener Append(Listener two)
    {
      Listener one = this;
      return new Listener { Unlisten = () => { one.Unlisten(); two.Unlisten(); } };
    }
  }
}
